### ISSUE TITLE
https request failing for PHP 5.6 with peer verification enabled (openssl)

### ISSUE THREAD

**Issue by justjulian** [HaxeFoundation/haxe#4581]
Since PHP 5.6 the openssl extension enables peer verification per default (http://php.net/manual/en/migration56.openssl.php). This causes https requests to fail as function connect in Socket.class.php uses IP address to connect to host instead of the hostname.

Example code:
`var https = new Http("https://s3-eu-central-1.amazonaws.com");`
`https.request(false);`

Will throw exception in Http.class.php with error message:
stream_socket_client(): Peer certificate CN='_.s3.eu-central-1.amazonaws.com' did not match expected CN='54.231.193.49' (errno: 2) in [...]\lib\sys\net\Socket.class.php at line #34stream_socket_client(): Peer certificate CN='_.s3.eu-central-1.amazonaws.com' did not match expected CN='54.231.193.49'

Also discussed here http://stackoverflow.com/questions/33043617/haxe-php-https-call-stream-socket-client-peer-certificate-cn-did-not-match/33058922#33058922 and here https://groups.google.com/forum/#!msg/haxelang/JMSBpF2sOi0/QpoNEjk9EQAJ

**Comment by yar3333** [comment]
This is a Neko problem too. I had to create SSL Socket manually and make a custom request.

**Comment by MatthijsKamstra** [comment]
Hi @justjulian ,
I have run into the same problem twice now... But this time I can't revert to an older PHP version.
And I can't really wait to 3.4 is released

In your original post you hint you found a solution.
Could you share it with me, I've been looking at this from all angles, but I can't find the solution in the Http.hx file :sob:

**Comment by justjulian** [comment]
Hi @MatthijsKamstra 
Shortly after I encountered this bug (and some others), I switched my project to Python. So, I don't remember exactly what I did half a year ago, but by going through my depreciated code now, it looks like that I wasn't happy with modifying Http.hx (I cannot find a custom Http.hx in my repository) and therefore probably downgraded my PHP version. Maybe I also just modified the generated PHP classes, which is of course a very bad solution. Sorry to disappoint you.

Though, I found one interesting comment in my code:
`// Wait until bug is resolved: https://github.com/HaxeFoundation/haxe/issues/4581`
`// Or create own socket and pass it in customRequest`
`var bytesOutput = new haxe.io.BytesOutput();`
`this.sigV4request.customRequest(true, bytesOutput, "POST");`

I didn't want to play around with sockets but maybe it is possible to use it as workaround.
